lib/plugins/paragon-webpack-plugin/ParagonWebpackPlugin.js

```diff
@@ -0,0 +1,184 @@
+const { Compilation, sources } = require('webpack');
+const parse5 = require('parse5');
+const {
+  getParagonVersion,
+  getParagonThemeCss,
+  replacePeriodsWithHyphens,
+} = require('../../../config/data/paragonUtils');
+
+const paragonVersion = getParagonVersion(process.cwd());
+const paragonThemeCss = getParagonThemeCss(process.cwd());
+
+class ParagonWebpackPlugin {
```

Comment:
Might want a class comment here that just describes what this plugin's job is.  I'm assuming it's adding a script with the `PARAGON` variable in it, but it's a lot of fairly dense webpack plugin code and I'm not sure if I'm missing something. :)